odoo/fields.py

```diff
@@ -457,18 +457,17 @@ def _get_attrs(self, model_class, name):
                 warnings.warn(f"precompute attribute has no impact on non stored field {self}")
                 attrs['precompute'] = False
         if attrs.get('company_dependent'):
-            # by default, company-dependent fields are not stored, not computed
-            # in superuser mode and not copied
-            attrs['store'] = False
-            attrs['compute_sudo'] = attrs.get('compute_sudo', False)
+            # if attrs.get('required'):
+            #     warnings.warn(f"company_dependent field {self} cannot be required")
+            # if attrs.get('translate'):
+            #     warnings.warn(f"company_dependent field {self} cannot be translated")
+            # if attrs.get('default')"
+            #     warnings.warn(f"company_dependent field {self} should not have default")
             attrs['copy'] = attrs.get('copy', False)
-            attrs['default'] = attrs.get('default', self._default_company_dependent)
-            attrs['compute'] = self._compute_company_dependent
-            if not attrs.get('readonly'):
-                attrs['inverse'] = self._inverse_company_dependent
-            attrs['search'] = self._search_company_dependent
-            attrs['depends_context'] = attrs.get('depends_context', ()) + ('company',)
-
+            # speed up search and on delete
+            attrs['index'] = attrs.get('index', 'btree_not_null')
+            attrs['prefetch'] = attrs.get('prefetch', 'company_dependent')
```

Comment:
Sine you have a custom prefetch value, you still have 2 queries if you read a company_dependent field and a normal field. I think you should not set `prefetch`.

Reply:
This is something to be discussed.
The main idea is to reduce the queries for UI which will call web_read and know which fields to fetch.

And yes we can prefetch all of them, but in this implementation, fetching company dependent fields need to know the `env.company` which may cause one extra query for `env.user.company_id` which is not necessary for some small simple non-company_dependent relevant rpcs.

Reply:
I was thinking about a simple use in the code, it would mean that if I access a field and a company-dependent field on the same model, I would have 2 queries. Shouldn't it be `env.company` which is accessed, and that one is used quite often, so probably already in cache.